### ISSUE TITLE
key_size in GifClass.h

### ISSUE THREAD

**Issue by marcmerlin** [moononournation/Arduino_GFX#26]
Hello again,
First, when you get a chance, try to recompile all your code with all warnings enabled in the arduino IDE (preferences/settings/compiler warnings/all)

this one is a compilation error on ESP32:
```
GifClass.h:417:59: error: narrowing conversion of 'key' from 'int32_t {aka int}' to 'uint8_t {aka unsigned char}' inside { } [-Werror=narrowing]
             table->entries[key] = (gd_Entry){1, 0xFFF, key};
```
I can change line 417 to say
        for (uint8_t key = 0; key < (1 << key_size); key++)

but I'm not too sure if key_size is smaller than 8, it looks like maybe not.

line 503, I see:
        gif_buf_read(gif->fd, &byte, 1);
        key_size = (int16_t)byte;

If it's a byte, why use (int16_t), shouldn't this be uint8_t ?

Later down, I see
        clear = 1 << key_size;

with
        uint16_t key, clear, stop;

So key is actually 16bits instead of 8.  and 1 << key_size likely is 16 bits and not 8 bits.

So maybe suffix in gd_Entry should be 16 bits?
```
typedef struct gd_Entry
{
    int32_t length;
    uint16_t prefix;
    uint8_t suffix;
} gd_Entry;
```

**Comment by marcmerlin** [comment]
other compilation warnings (those are warnings, but still probably a good idea to fix)
```
Arduino_ESP32SPI_DMA.cpp: In member function 'virtual void Arduino_ESP32SPI_DMA::begin(int32_t, int8_t)':
Arduino_ESP32SPI_DMA.cpp:87:3: warning: missing initializer for member 'spi_bus_config_t::flags' [-Wmissing-field-initializers]
   };

   ^
Arduino_ESP32SPI_DMA.cpp:87:3: warning: missing initializer for member 'spi_bus_config_t::intr_flags' [-Wmissing-field-initializers]
Arduino_ESP32SPI_DMA.cpp:101:3: warning: missing initializer for member 'spi_device_interface_config_t::pre_cb' [-Wmissing-field-initializers]
   };

   ^
Arduino_ESP32SPI_DMA.cpp:101:3: warning: missing initializer for member 'spi_device_interface_config_t::post_cb' [-Wmissing-field-initializers]
```

**Comment by marcmerlin** [comment]
my apologies, you've already fixed this in 55b2adb8acee86f6318a9ce344fbc4102e44c469

**Comment by marcmerlin** [comment]
if you care, though, this warning is still there:
```
Arduino_ESP32SPI_DMA.cpp: In member function 'virtual void Arduino_ESP32SPI_DMA::begin(int32_t, int8_t)':
Arduino_ESP32SPI_DMA.cpp:87:3: warning: missing initializer for member 'spi_bus_config_t::flags' [-Wmissing-field-initializers]
   };

   ^
Arduino_ESP32SPI_DMA.cpp:87:3: warning: missing initializer for member 'spi_bus_config_t::intr_flags' [-Wmissing-field-initializers]
Arduino_ESP32SPI_DMA.cpp:101:3: warning: missing initializer for member 'spi_device_interface_config_t::pre_cb' [-Wmissing-field-initializers]
   };

   ^
Arduino_ESP32SPI_DMA.cpp:101:3: warning: missing initializer for member 'spi_device_interface_config_t::post_cb' [-Wmissing-field-initializers]
```
